src/odemis/driver/focustracker.py

```diff
@@ -39,171 +44,177 @@ def __init__(self, name, role, channel, node_idx, **kwargs):
         model.HwComponent.__init__(self, name, role, **kwargs)
         # Connect to the CANbus and the CANopen network.
         self.network = canopen.Network()
-        self.network.connect(bustype='socketcan', channel=channel)
+        bustype = 'socketcan' if channel != 'fake' else 'virtual'
+        try:
+            self.network.connect(bustype=bustype, channel=channel)
+        except IOError as exp:
+            if exp.errno == 19:
+                raise HwError("Focus Tracker is not connected.")
+            else:
+                raise
         self.network.check()
-        self.node = canopen.RemoteNode(node_idx, os.path.dirname(odemis.__file__) + "/driver/FocusTracker.eds")
+        object_dict = pkg_resources.resource_filename("odemis.driver", "FocusTracker.eds")
+        if channel == 'fake':
+            self.node = FakeRemoteNode(node_idx, object_dict)
+        else:
+            self.node = canopen.RemoteNode(node_idx, object_dict)
         self.network.add_node(self.node)
+        self._swVersion = "python-canopen v%s , python-can v%s" % (canopen.__version__, can.__version__)
+        self._hwVersion = self.node.sdo["Identity Object"]["Revision number"].raw
```

Comment:
Revision number is an int, but hwVersion should be a string. Also, i wonder if there is some official encoding of the revision number, because it's currently set to 65537, which seems an awful lot... maybe it's "1.1"?

Reply:
65537 is 0x10001 so I guess it's similar to 1.1, I'll set hwVersion as a hex string; `hwVersion = '0x10001'`